src/subscription_manager/cli_command/environments.py

```diff
@@ -77,3 +91,101 @@ def _do_command(self):
+                )
+            )
+            print(_("Environments updated."))
+        else:
+            system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support environment updates."))
+
+    def _list_environments(self):
+        """
+        List the environments based on the option selected in the command line
+        enabled/disabled/all
+        """
+        if not self.cp.has_capability(MULTI_ENV) and (self.options.enabled or self.options.disabled):
+            system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support multi-environment operations."))
+        environments = []
+        if self.options.enabled:
+            environments = self.cp.getConsumer(self.identity.uuid)['environments']
+        else:
+            org_environments = self._get_environments(self.org)
+            if self.options.disabled:
+                consumer_id_list = []
+                for env in self.cp.getConsumer(self.identity.uuid)['environments']:
+                    consumer_id_list.append(env['id'])
+                for env in org_environments:
+                    if env['id'] not in consumer_id_list:
+                        environments.append(env)
+            else:
+                environments = org_environments
+
+        if len(environments):
+            print("+-------------------------------------------+")
+            print("          {env}".format(env=_('Environments')))
+            print("+-------------------------------------------+")
+            for env in environments:
+                print(columnize(ENVIRONMENT_LIST, echo_columnize_callback, env['name'],
+                                env['description'] or "") + "\n")
+        else:
+            print(_("This list operation does not have any environments to report."))
+
+    def _process_environments(self, admin_cp, owner_key, options):
+        """
+        Gather environment list from server and pass to method to
+        validate the environment input against it
+        """
+        all_env_list = admin_cp.getEnvironmentList(owner_key)
+        return check_set_environment_names(all_env_list, options.set)
+
+    @property
+    def org(self):
+        """
+        An override is needed here because we need to use the org from the
+        identity for this command if this system is already registered
+        """
+        self.identity = require(IDENTITY)
+        if self.identity.is_valid():
+            self._org = self.cp.getOwner(self.identity.uuid)['key']
+            return self._org
+        elif self.options.org:
+            self._org = self.options.org
+            return self._org
+        else:
+            if not self.options.username:
+                self.options.username = self._username
+            return super().org
+
+
+def check_set_environment_names(all_env_list, name_string):
+    """
+    Checks the environment name(s) input for duplicates and
+    inclusion in the full list
+    """
+    names = name_string.split(",")
```

Comment:
Do we want to have the more reliable? I believe that following command is correct, but sub-man complains:

```
[root@thinkpad-t580 subscription-manager]# PYTHONPATH=./src python -m \
     subscription_manager.scripts.subscription_manager \
     environments --username admin --password admin --set "env-name-1,env-name-2"
Environments updated.

[root@thinkpad-t580 subscription-manager]# PYTHONPATH=./src python -m \
     subscription_manager.scripts.subscription_manager \
    environments --username admin --password admin --set "env-name-1, env-name-2" \
    # Note: there is space between comma and last name
No such environment:  env-name-2
```

I would modify `names` list with something like this:

```python
names = [name.strip() for name in names]
```